File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer, Integer> map = new HashMap<>();

        int majority = 0;
        int res = 0;

        for(int n: nums)
        {
            map.put(n, map.getOrDefault(n, 0) + 1);

            if(map.get(n) > majority)
            {
                res = n;
                majority = map.get(n);
            }
        }

        return res;
    }
}

File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int ans, majority = 0;

        for (int num : nums) {
            if (majority == 0) { // 表示当前没有候选人
                ans = num;       // 更新新的候选人
            }
            majority += ans == num ? 1 : -1; 
        }

        return ans;
    }
};